src/test/java/swinglib/JButtonBuilderTest.java

```diff
@@ -0,0 +1,70 @@
+package swinglib;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import java.awt.event.ActionEvent;
+import java.util.Arrays;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.swing.JButton;
+
+import org.hamcrest.core.Is;
+
+import org.junit.Test;
+
+public class JButtonBuilderTest {
+
+  @Test
+  public void checkTitle() {
+    final String value = "testing title";
+    final JButton button = JButtonBuilder.builder()
+        .title(value)
+        .actionListener(() -> {
+        })
+        .build();
+    assertThat(button.getText(), Is.is(value));
+  }
+
+  @Test
+  public void checkActionListener() {
+    // button action will be to add one to our integer, we'll fire the button action and verify we get the +1
+    final AtomicInteger integer = new AtomicInteger(0);
+    final JButton button = JButtonBuilder.builder()
+        .title("title")
+        .actionListener(integer::incrementAndGet)
+        .toolTip("toolTip")
+        .build();
+
+    Arrays.stream(button.getActionListeners())
+        .forEach(listener -> listener.actionPerformed(new ActionEvent(new Object(), 0, "")));
+    assertThat(integer.get(), Is.is(1));
+
+    assertThat(button.getToolTipText(), Is.is("toolTip"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void titleIsRequired() {
```

Comment:
The latest change looks good, and the test names are very clear. :+1:

I still have a small concern about the body of `titleCannotBeEmpty()`.  If we only care about the behavior of `title()`, I find having to call all the additional setters required by `build()` to be a bit distracting.  In addition, the test could falsely pass if, for example, `title()` didn't throw but `actionListener()` did.

I could be missing some intent here, but is the following test equivalent?

```java
@Test(expected = IllegalArgumentException.class)
public void titleCannotBeEmpty() {
  JButtonBuilder.builder().title("");
}
```

If so, it would save some lines, and the next dev to come along and add a new builder property wouldn't think they had to go back and retrofit all the existing setter tests with calls to their newly-added setter.  IMO, only `build()` tests should require calls to setters for all required properties; the setter tests themselves should be focused on, well, themselves. :smile: